plugins/search-backend/src/service/router.ts

```diff
@@ -129,7 +132,10 @@ export async function createRouter(
   const router = Router();
   router.get(
     '/query',
-    async (req: express.Request, res: express.Response<SearchResultSet>) => {
+    async (
+      req: express.Request,
+      res: express.Response<SearchResultSet | string>,
```

Comment:
I've never used this before, but I'd be interested to know:

Is the end-user experience any better (compared to what you showed) if you return an object of type [`ErrorResponseBody`](https://backstage.io/docs/reference/errors.errorresponsebody) instead of just a string?